src/modules/flight_mode_manager/tasks/AutoFollowTarget/FlightTaskAutoFollowTarget.hpp

```diff
@@ -130,81 +141,123 @@ class FlightTaskAutoFollowTarget : public FlightTask
 
-	// Follow angle is defined with 0 degrees following from front, and then clockwise rotation
-	float _follow_angle_rad{0.0f};
-	AlphaFilter<float> _follow_angle_filter;
+	// Sticks object to read in stick commands from the user
+	Sticks _sticks;
 
 	// Estimator for target position and velocity
+	TargetEstimator _target_estimator;
 	follow_target_estimator_s _follow_target_estimator;
-	matrix::Vector2f _target_velocity_unit_vector;
 
-	// Lowpass filters for smoothingtarget position because it's used for setpoint generation
-	AlphaFilter<matrix::Vector3f> _target_position_filter;
+	// Last target estimator timestamp to handle timeout filter reset
+	uint64_t _last_valid_target_estimator_timestamp{0};
+
+	// Second Order Filter to calculate kinematically feasible target position
+	SecondOrderReferenceModel<matrix::Vector3f> _target_pose_filter;
+
+	// Internally tracked Follow Target characteristics, to allow RC control input adjustments
+	float _follow_target_distance{0.0f};
+	float _follow_target_height{0.0f};
+	float _follow_angle_rad{0.0f}; // 0 degrees following from front, and then clockwise rotation
+
+	// Estimated (Filtered) target orientation setpoint
+	float _target_orientation_rad{0.0f};
+
+	// Orbit angle setpoint, measured in global frame, against the target
+	float _orbit_angle_setpoint{0.0f};
+
+	// Actual drone to target heading [rad]
+	float _drone_to_target_heading{0.0f};
+
+	// Actual orbit angle in relation to the target
+	float _measured_orbit_angle{0.0f};
 
-	// Lowpass filter for smoothing the offset vector and have more dynamic shots when target changes direction
-	AlphaFilter<matrix::Vector2f> _offset_vector_filter;
+	// Actual drone to target 2d position vector
+	Vector2f _drone_to_target_vector{0.0f, 0.0f};
 
-	// Lowpass filter assuming  values 0-1, for avoiding big steps in velocity feedforward
-	AlphaFilter<float> _velocity_ff_scale;
+	// Tracked orbit tangential speed, to compensate for the orbital motion for velocity setpoints
+	Vector2f _orbit_tangential_velocity{0.0f, 0.0f};
 
 	// NOTE: If more of these internal state variables come into existence, it
 	// would make sense to create an internal state machine with a single enum
 	bool _emergency_ascent = false;
 
-	DEFINE_PARAMETERS(
-		(ParamFloat<px4::params::NAV_MIN_FT_HT>) _param_nav_min_ft_ht,
+	// Yaw setpoint filter to remove jitter-ness
+	AlphaFilter<float> _yaw_setpoint_filter;
+
+	// Variable to remember the home position's z coordinate, which will be baseline for the position z setpoint
+	float _home_position_z{0.0f};
+
+	float _gimbal_pitch{0};
+
+	DEFINE_PARAMETERS_CUSTOM_PARENT(
+		FlightTask,
+		(ParamFloat<px4::params::NAV_FT_HT>) _param_nav_ft_ht,
 		(ParamFloat<px4::params::NAV_FT_DST>) _param_nav_ft_dst,
 		(ParamInt<px4::params::NAV_FT_FS>) _param_nav_ft_fs,
 		(ParamInt<px4::params::NAV_FT_ALT_M>) _param_nav_ft_alt_m,
-		(ParamInt<px4::params::NAV_FT_GMB_M>) _param_nav_ft_gmb_m,
```

Comment:
Is the gimbal tracking mode setting no longer required? What's the tracking mode now?